File: src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author aulas
 */
public class Shield {
    private float protection;
    private int uses;
    
    public Shield(float protection,int uses){
        this.protection = protection;
        this.uses = uses;
    }
    
    public float protect(){
        if(uses > 0){
            uses--;
            return protection;
        }else{
            return uses;
        }
    }
    @Override
    public String toString(){
        return "S[" + protection + "," + uses + "]";
    }
    
    public boolean discard(){
        return Dice.discardElement(uses);
    }
}

File: RubyIrrgarten/Irrgarten1/Shield.rb
require_relative 'Dice'
module Irrgarten
  class Shield
    def initialize(protection,uses)
      @protection = protection
      @uses = uses
    end

    def protect
      if @uses > 0
       @uses -= 1
       @protection.to_f
      else
        @uses.to_f
      end
    end

    def to_s
      "W[" + @protection.to_s + "," + @uses.to_s + "]"
    end

    def discard
      Dice.discard_element(@uses)
    end
  end
end